### PR TITLE
Add authenication logic

### PR DESCRIPTION
**Pull request by darenang44**

### CODE REVIEW

File: client/app/auth/auth.js [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
 
-.controller('AuthController', function($scope, $http){
-  $scope.signup = 'Signup';
-  $scope.signin = 'Signin';
+.controller('AuthController', function ($scope, $http, $window){
+  $scope.user = {username: '', password: ''};
+  $scope.message = '';
+  $scope.submit = function () {
+    $http.post('/authenticate', $scope.user)
+    .success( function (data){
```
Comment (dting) [inline client/app/auth/auth.js]: extra space and missing space

File: client/app/auth/auth.js [hunk 2]
```diff
@@ -1,6 +1,17 @@
 angular.module('throwcast.auth', [])
 
-.controller('AuthController', function($scope, $http){
-  $scope.signup = 'Signup';
-  $scope.signin = 'Signin';
+.controller('AuthController', function ($scope, $http, $window){
+  $scope.user = {username: '', password: ''};
+  $scope.message = '';
+  $scope.submit = function () {
```
Comment (dting) [inline client/app/auth/auth.js]: This line doesn't seem to have consistent spacing.

File: client/app/auth/service.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  return {
+    request: function (config) {
+      config.headers = config.headers || {};
+      if ($window.sessionStorage.token) {
+        config.headers.Authorization = 'Bearer ' + $window.sessionStorage.token;
+      }
+      return config;
+    },
+    response: function (rejection) {
+      if (response.status === 401) {
```
Comment (dting) [inline client/app/auth/service.js:13]: I think we redirect to login on a rejection with response 401. Take a look online on how to handle that status code.

File: client/app/auth/service.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        $log('Unauthorized User');
```
Comment (dting) [inline client/app/auth/service.js]: Is `$log` global? Doesn't look injected.

File: client/app/auth/service.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      }
+      return $q.when(rejection);
+    }
+  };
+})
+.config(function ($httpProvider) {
```
Comment (dting) [inline client/app/auth/service.js:21]: Missing a new line above this one if you want to be consistent with the extra lines you have between the module declaration and the chained calls.

File: client/app/auth/service.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+angular.module('throwcast.auth', [])
+
+.factory('authInterceptor', function ($q, $window){
```
Comment (dting) [inline client/app/auth/service.js]: missing space.

File: client/app/auth/auth.js [hunk 7]
```diff
@@ -1,6 +1,17 @@
 angular.module('throwcast.auth', [])
 
-.controller('AuthController', function($scope, $http){
-  $scope.signup = 'Signup';
-  $scope.signin = 'Signin';
+.controller('AuthController', function ($scope, $http, $window){
```
Comment (dting) [inline client/app/auth/auth.js]: missing space

**Comment by dting** [comment]
LGTM.

File: client/app/auth/auth.js (hunk 1 above)
Comment (darenang44) [inline client/app/auth/auth.js]: Done

File: client/app/auth/service.js (hunk 3 above)
Comment (darenang44) [inline client/app/auth/service.js:13]: Done, used $location.path('/signin');

File: client/app/auth/service.js (hunk 4 above)
Comment (darenang44) [inline client/app/auth/service.js]: redirected to signin page

File: client/app/auth/service.js (hunk 5 above)
Comment (darenang44) [inline client/app/auth/service.js:21]: Done

File: client/app/auth/service.js (hunk 6 above)
Comment (darenang44) [inline client/app/auth/service.js]: Done

File: client/app/auth/auth.js (hunk 7 above)
Comment (darenang44) [inline client/app/auth/auth.js]: Done

**Pull request merged**